### ISSUE TITLE
[Transfer Learning] Pipcook is supposed to support transfer learning with plugin specification 

### ISSUE THREAD

**Issue by rickycao-qy** [alibaba/pipcook#689]
Pipcook is supposed to standardize how the plugin supports transfer learning.

Currently, we provide an interface for model-define plugin with `modelPath` attribute and leaves other works in plugins. The plugin has full freedom to decide how and whether to support transfer learning. This will cause confusion to users and is hard to uniform this ability.

I suggest we should go further to define how to do transfer learning in pipcook and force plugins to follow the standard.

Do you have more thoughts on this. Will look forward to your opinions.

**Comment by upupzealot** [comment]
Will all model-define plugins have this param?

**Comment by upupzealot** [comment]
Let's list all model-define plugins and see which already has this param

- [x] @pipcook/plugins-tensorflow-mobilenet-model-define
- [x] @pipcook/plugins-tensorflow-resnet-model-define
- [x] @pipcook/plugins-detectron-fasterrcnn-model-define
- [x] @pipcook/plugins-bayesian-model-define
- [ ] @pipcook/plugins-pytorch-yolov5-model-define
- [ ] @pipcook/plugins-fasttext-model-define
- [ ] @pipcook/plugins-tensorflow-cycle-gan-model-define
- [x] @pipcook/plugins-detectron-maskrcnn-model-define
- [ ] @pipcook/plugins-tensorflow-bert-ner-model-define
- [x] @pipcook/plugins-tfjs-simplecnn-model-define
- [ ] @pipcook/plugins-tfjs-text-lstm-model-define
- [ ] @pipcook/plugins-tfjs-text-lstm-model-define

**Comment by upupzealot** [comment]
For me currently, I can image serval params that maybe help in transfer learning.

- recoverPath: as mentioned above, specified the directory which contains the "model" directory (which can be found in the output)
- epochs: how many epochs will train for current job (currently defined in model-train plugins)
- maxEpoch: job will be stopped when it reaches maxEpoch

**Comment by rickycao-qy** [comment]
> Will all model-define plugins have this param?

It's likely some of models cannot support transfer learning. We can think of a way that Pipcook could be infomred which one support and which one does not support

**Comment by rickycao-qy** [comment]
> For me currently, I can image serval params that maybe helpful in transfer learning.
> 
> * recoverPath
>   
>   * as mentioned above, specified the directory which contains the "model" directory
>   * which can be found in the output
> * freeze
>   
>   * whether to freeze first several layers
>   * currently a boolean flag, not sure if we can make this an integer, to specify how many layers to freeze
>   * 0 could mean not to freeze
> * labelMap
>   
>   * a string yo string map(object), to descption the describe the mapping relations for new labels
>   * not necessary
> * epochs (currently defined in model-train plugins)
>   
>   * how many epochs will train for current job
> * maxEpoch (could be in model-train plugins also)
>   
>   * job will be stopped when it reaches maxEpoch

That sounds good. We need to split which methods are used in model-define and which methods are used in model-train. I guess model-train do not need to know whether currently it's transfer learning or new-model learning.

**Comment by upupzealot** [comment]
"freeze (fine-tuning)" params also reminds me that, maybe we need some extra params in "model-define" plugins.
That will make the conception of "layer" complete : )
